firebase-perf/src/main/java/com/google/firebase/perf/injection/modules/GaugeCollectorsModule.java

```diff
@@ -0,0 +1,62 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.perf.injection.modules;
+
+import static android.system.Os.sysconf;
+
+import android.os.Build;
+import android.system.OsConstants;
+import dagger.Module;
+import dagger.Provides;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+
+/**
+ * Provider for {@link com.google.firebase.perf.gauges}.
+ *
+ * @hide
+ */
+@Module
+public class GaugeCollectorsModule {
+
+  public static final long INVALID_CPU_COLLECTION_FREQUENCY = -1;
+
+  @Provides
+  ScheduledExecutorService providesScheduledExecutorService() {
+    return Executors.newSingleThreadScheduledExecutor();
+  }
+
+  @Provides
+  String providesProcFileName() {
+    int pid = android.os.Process.myPid();
+    String procFileName = "/proc/" + Integer.toString(pid) + "/stat";
```

Comment:
Usually, it's preferable to use String.format over string concatenation. It makes the strings easier to read, and provides some compile-time checking with typing (since you could do %d for the `pid` instead of converting it to a string).

Reply:
Done.